src/storage/disk_table.cc

```diff
@@ -1097,8 +1158,138 @@ void DiskTableRowIterator::SeekToFirst() {
+            const auto& col = cols->Get(idx);
+            schema.emplace(col.name(), ColumnDef(col.name(), idx, col.data_type(), col.not_null()));
+        }
+
+        std::vector<ColumnDef> col_vec;
+        for (const auto& col_name : column_key.col_name()) {
+            auto it = schema.find(col_name);
+            if (it == schema.end()) {
+                PDLOG(WARNING, "not found col_name[%s]. tid %u pid %u", col_name.c_str(), id_, pid_);
+                return false;
+            }
+            col_vec.push_back(it->second);
+        }
+
+        uint32_t inner_id = table_index_.GetAllInnerIndex()->size();
+
+        index_def = std::make_shared<IndexDef>(column_key.index_name(), table_index_.GetMaxIndexId() + 1,
+                IndexStatus::kReady, ::openmldb::type::IndexType::kTimeSerise, col_vec);
+
+
+        if (!column_key.ts_name().empty()) {
+            auto ts_iter = schema.find(column_key.ts_name());
+            if (ts_iter == schema.end()) {
+                PDLOG(WARNING, "not found col_name[%s]. tid %u pid %u", column_key.ts_name(), id_, pid_);
+                return false;
+            }
+            index_def->SetTsColumn(std::make_shared<ColumnDef>(ts_iter->second));
+        } else {
+            index_def->SetTsColumn(std::make_shared<ColumnDef>(DEFUALT_TS_COL_NAME, DEFUALT_TS_COL_ID,
+                        ::openmldb::type::kTimestamp, true));
+        }
+
+        if (column_key.has_ttl()) {
+            index_def->SetTTL(::openmldb::storage::TTLSt(column_key.ttl()));
+        } else {
+            index_def->SetTTL(*(table_index_.GetIndex(0)->GetTTL()));
+        }
+
+        if (table_index_.AddIndex(index_def) < 0) {
+            PDLOG(WARNING, "add index failed. tid %u pid %u", id_, pid_);
+            return false;
+        }
+
+        index_def->SetInnerPos(inner_id);
+        std::vector<std::shared_ptr<IndexDef>> index_vec = {index_def};
+        auto inner_index_st = std::make_shared<InnerIndexSt>(inner_id, index_vec);
+        table_index_.AddInnerIndex(inner_index_st);
+        table_index_.SetInnerIndexPos(new_table_meta->column_key_size() - 1, inner_id);
+
+        rocksdb::ColumnFamilyOptions cfo;
+
+        if (storage_mode_ == ::openmldb::common::StorageMode::kSSD) {
+            cfo = rocksdb::ColumnFamilyOptions(ssd_option_template);
+        } else {
+            cfo = rocksdb::ColumnFamilyOptions(hdd_option_template);
+        }
+
+        cfo.comparator = &cmp_;
+        cfo.prefix_extractor.reset(new KeyTsPrefixTransform());
+
+        if (index_def->GetTTLType() == ::openmldb::storage::TTLType::kAbsoluteTime ||
+            index_def->GetTTLType() == ::openmldb::storage::TTLType::kAbsOrLat) {
+            cfo.compaction_filter_factory = std::make_shared<AbsoluteTTLFilterFactory>(inner_index_st);
+        }
+
+        rocksdb::ColumnFamilyHandle* handle;
+        cf_ds_.push_back(rocksdb::ColumnFamilyDescriptor(index_def->GetName(), cfo));
```

Comment:
this one also need the `lock` to protect?

Reply:
Because except for some initialization functions, only here appears `cf_ds_`. If we don't consider concurrent calls to `Addindex`, it looks fine not to protect `cf_ds_`.
If we have to, yes, definitely and more than so.

Reply:
`Addindex` may be called concurrently. We can use same lock to protect it.